docs/toolhive/guides-vmcp/authentication.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: Authentication
+description: Configure client and backend authentication for Virtual MCP Server.
+---
+
+Virtual MCP implements a two-boundary authentication model that separates client
+authentication from backend authentication, giving you centralized control over
+access while supporting diverse backend requirements.
+
+## Two-boundary authentication model
+
+```mermaid
+flowchart LR
+    subgraph Boundary1[" "]
+        direction TB
+        Client[MCP Client]
+        B1Label["**Boundary 1**<br>Client → Virtual MCP"]
+    end
+
+    subgraph vMCP[Virtual MCP Server]
+        Auth[Token Validation]
+        Exchange[Token Exchange]
+    end
+
+    subgraph Boundary2[" "]
+        direction TB
+        B2Label["**Boundary 2**<br>Virtual MCP → Backend APIs"]
+        GitHub[GitHub API]
+        Jira[Jira API]
+    end
+
+    Client -->|"Token: aud=vmcp"| Auth
+    Auth --> Exchange
+    Exchange -->|"Token: aud=github-api"| GitHub
+    Exchange -->|"Token: aud=jira-api"| Jira
+```
+
+**Boundary 1 (Incoming):** Clients authenticate to Virtual MCP using a token
+with `aud=vmcp`. This is your organization's identity layer.
+
+**Boundary 2 (Outgoing):** Virtual MCP exchanges the client token for
+backend-specific tokens. Each backend API receives a token scoped to its
+requirements.
+
+## Incoming authentication
+
+Configure how clients authenticate to Virtual MCP.
+
+### Anonymous (development only)
+
+No authentication required. Use only for local development.
+
+```yaml
+spec:
+  incomingAuth:
+    type: anonymous
+```
+
+### OIDC authentication
+
+Validate tokens from an external identity provider.
+
+{/* TODO: Add OIDC configuration example */}
+
+### Kubernetes service account tokens
+
+Authenticate using Kubernetes service account tokens for in-cluster clients.
+
+{/* TODO: Add Kubernetes auth configuration example */}
+
+## Outgoing authentication
+
+Configure how Virtual MCP authenticates to backend APIs.
+
+### Discovery mode (recommended)
+
+Automatically discover authentication configuration from each backend's
+MCPServer resource:
```

Comment:
I think it would be nice to say what exactly does this mean - that the back end authentication will be discovered based on the MCPServer's MCPExternalAuth CR but if there's none, then vMCP will be using an anonymous connection to the back end service.